### PR TITLE
Restored wreck logging. Closes #469.

### LINKED ISSUE #469: Restore Wreck Logging
Now that https://github.com/hapijs/wreck/issues/111 has been resolved, we can put back wreck logging. Allow for this to be disabled as well since I think most people won't use this feature.

### PR DESCRIPTION
**Pull request by arb**
Cleaned up event handling logic and just added a boolean to control whether events are reported on not. It was much easier and required far less code to keep a handler to all the added event listeners.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,8 @@
     "async": "1.5.x",
     "code": "3.x.x",
     "hapi": "15.x.x",
-    "lab": "11.x.x"
+    "lab": "11.x.x",
+    "wreck": "9.x.x"
```
Comment (vdeturckheim) [inline package.json]: Wreck is 10 now
https://github.com/hapijs/wreck/issues/150

**Comment by kevinmstephens** [comment]
@Marsup Just a friendly request to complete the review if you can fit it in. Waiting on this release to hit NPM. Cheers.

File: package.json [hunk 2]
```diff
@@ -28,11 +28,11 @@
     "async": "1.5.x",
     "code": "3.x.x",
     "hapi": "15.x.x",
-    "lab": "11.x.x"
+    "lab": "11.x.x",
+    "wreck": "10.x.x"
   },
   "scripts": {
-    "test": "lab -m 5000 -t 100 -v -La code",
-    "test-cov-html": "lab -m 5000 -r html -o coverage.html -La code"
```
Comment (Marsup) [inline package.json]: Mistake ?

File: test/monitor.js [hunk 3]
```diff
@@ -912,5 +920,60 @@ describe('Monitor', () => {
...
+                    }, (res) => {
+
+                        expect(res.statusCode).to.equal(200);
+                        // account for hapi15 differences
+                        expect(out.data.length).to.be.greaterThan(0);
+                        const result = out.data.find((event) => {
+
+                            return event.event === 'wreck';
+                        });
+                        expect(result).to.be.an.instanceof(Utils.WreckResponse);
```
Comment (Marsup) [inline test/monitor.js]: Shouldn't you check what's in it as well ?

File: API.md [hunk 4]
```diff
@@ -192,6 +193,9 @@ Event object associated with the `responseEvent` event option into Good. `reques
 - `route` - route path used by request. Maps to `request.route.path`.
 - `log` - maps to `request.getLog()` of the hapi request object.
 - `config` - plugin-specific config object combining `request.route.settings.plugins.good` and `request.plugins.good`. Request-level overrides route-level. Reporters could use `config` for additional filtering logic.
+- `headers` - the request headers if `includes.request` includes "headers"
+- `requestPayload` - the request payload if `includes.request` includes "payload"
```
Comment (Marsup) [inline API.md]: Those things don't appear in code anywhere, should it ?

File: lib/monitor.js [hunk 5]
```diff
@@ -134,57 +124,58 @@ class Monitor {
...
+                const args = Array(arguments.length);
+                for (let j = 0; j < arguments.length; ++j) {
+                    args[j] = arguments[j];
+                }
+                const payload = {
+                    event,
+                    timestamp: Date.now(),
+                    payload: args
+                };
+                self.push(() => Object.assign({}, payload));
```
Comment (arb) [inline lib/monitor.js]: Yes. It's as close as I'm willing to do for something that gets called so often.

File: lib/utils.js [hunk 6]
```diff
@@ -143,6 +143,40 @@ class RequestLog {
...
+
+        response = response || {};
+        uri = uri || {};
+        this.event = 'wreck';
+        this.timestamp = Date.now();
+        this.timeSpent = this.timestamp - start;
+        this.pid = process.pid;
+
+        if (error) {
+            this.error = {
```
Comment (arb) [inline lib/utils.js]: Again, I'm just porting what was already there.

File: package.json (hunk 2 above)
Comment (arb) [inline package.json]: Nope. Never use any of those.

File: API.md (hunk 4 above)
Comment (arb) [inline API.md]: https://github.com/hapijs/good/blob/master/lib/utils.js#L93-L99

**Pull request merged**